pkg/lib/util/path_util.go

```diff
@@ -1,57 +1,51 @@
 package util
 
 import (
-	"regexp"
+	"fmt"
 	"strconv"
 	"strings"
 )
 
-/*
-throughout this file we assume 'manifestDir' to be an absolute path
-*/
+// PathSpec is a path specifier for including tests within manifests.
+type PathSpec struct {
+	// ParallelRuns matches the number of parallel runs specified
+	// in a path spec like "5@foo.json"
+	ParallelRuns int
 
-func IsPathSpec(pathSpec string) bool {
-	if len(pathSpec) < 3 {
-		return false
-	}
-	if strings.HasPrefix(pathSpec, "@") {
-		return true
-	}
-	if strings.HasPrefix(pathSpec, "p@") {
-		return true
-	}
-	// pathSpec could have trailing quotes
-	if strings.HasPrefix(pathSpec, "\"@") {
-		return true
-	}
-	if strings.HasPrefix(pathSpec, "\"p@") {
-		return true
-	}
-
-	return IsParallelPathSpec(pathSpec)
+	// Path matches the literal path, e.g. foo.json in "@foo.json"
+	Path string
 }
 
-func IsParallelPathSpec(pathSpec string) bool {
-	n, _ := GetParallelPathSpec(pathSpec)
-	return n > 0
-}
-
-func GetParallelPathSpec(pathSpec string) (parallelRepititions int, parsedPath string) {
-	regex := *regexp.MustCompile(`^\"{0,1}p(\d+)@(.+)\"{0,1}$`)
-	res := regex.FindAllStringSubmatch(pathSpec, -1)
+// ParsePathSpec tries to parse the given string into a PathSpec.
+//
+// The string takes the format "[n]@file.json". Invalid path specs
+// result in an error.
+func ParsePathSpec(s string) (*PathSpec, error) {
+	var ok bool
+	var err error
+	var parallelRuns string
+	var spec PathSpec
```

Comment:
Die Return Variablen bitte direkt in der func decl benennen, das erscheint dann in Godoc und unterstützt nochmal die Lesbarkeit. Auch macht das die std lib genauso. Auch wenn "spec" dann nochmal initialisiert werden muss `spec = &PathSpec{}`.

Und "var" mit mehreren Zeilen bitte als `var ( ... )` schreiben.

Reply:
Kann man in dem Fall vertreten (ohne names params). Aber bei anderen Fällen nicht. Z.B. aus der Standard Lib:

```
func makeASCIISet(chars string) (as asciiSet, ok bool)
func Cut(s, sep string) (before, after string, found bool)
```

Ohne die named params wären die nicht so lesbar.

Dann von mir aus so ladden, aber die "var (...)" änderung noch machen.